src/apps/chahub/models.py

```diff
@@ -78,37 +85,31 @@ def save(self, force_to_chahub=False, *args, **kwargs):
         if settings.CHAHUB_API_URL:
             is_valid = self.get_chahub_is_valid()
 
-            logger.info("ChaHub :: {}={} is_valid = {}".format(self.__class__.__name__, self.pk, is_valid))
+            logger.info(f"ChaHub :: {self.__class__.__name__}({self.pk}) is_valid = {is_valid}")
 
-            if is_valid and self.chahub_needs_retry and not force_to_chahub:
-                logger.info("ChaHub :: This has already been tried, waiting for do_retries to force resending")
-            elif is_valid:
-                # data = json.dumps(self.get_chahub_data())
+            if is_valid:
                 data = self.get_chahub_data()
                 data_hash = hashlib.md5(json.dumps(data).encode('utf-8')).hexdigest()
 
                 # Send to chahub if we haven't yet, we have new data
                 if not self.chahub_timestamp or self.chahub_data_hash != data_hash:
-                    resp = send_to_chahub(self.get_chahub_endpoint(), data)
+                    resp = send_to_chahub(self.get_chahub_endpoint(), data, update=bool(self.chahub_timestamp))
 
                     if resp and resp.status_code in (200, 201):
-                        logger.info("ChaHub :: Received response {} {}".format(resp.status_code, resp.content))
+                        logger.info(f"ChaHub :: Received response {resp.status_code} {resp.content}")
                         self.chahub_timestamp = timezone.now()
                         self.chahub_data_hash = data_hash
                         self.chahub_needs_retry = False
                     else:
-                        status = resp.status_code if hasattr(resp, 'status_code') else 'N/A'
-                        body = resp.content if hasattr(resp, 'content') else 'N/A'
-                        logger.info("ChaHub :: Error sending to chahub, status={}, body={}".format(status, body))
+                        status = getattr(resp, 'status_code', 'N/A')
+                        body = getattr(resp, 'content', 'N/A')
+                        logger.info(f"ChaHub :: Error sending to chahub, status={status}, body={body}")
```

Comment:
above is to just say `N/A` instead of `None`?

Reply:
Pretty much